icc-x-api/icc-patient-x-api.ts

```diff
@@ -575,12 +575,24 @@ export class IccPatientXApi extends iccPatientApi {
   }
 
   encrypt(user: models.UserDto, pats: Array<models.PatientDto>): Promise<Array<models.PatientDto>> {
+    const fixEncryptionKeys = (p: models.PatientDto) => {
+      if (p.delegations && p.delegations[(user.healthcarePartyId || user.patientId)!])
+        return this.initEncryptionKeysForDelegates(
+          user,
+          p,
+          Object.keys(p.delegations)
+        ) as Promise<any>
+      throw new Error(
```

Comment:
Just for readability, I would add the `else` (even though the if has a `return` statement). Or maybe even an 'early return' where you check the opposite and throw the error and in the end return the encryptionKeys.